picoquic/sockloop.c

```diff
@@ -109,7 +110,7 @@ static int udp_gso_available = 0;
 #endif
 #endif
 
-int picoquic_packet_loop_open_sockets(int local_port, int local_af, SOCKET_TYPE * s_socket, int * sock_af, 
+int picoquic_packet_loop_open_sockets(char *ip, int local_port, int local_af, SOCKET_TYPE * s_socket, int * sock_af,
     uint16_t * sock_ports, int socket_buffer_size, int nb_sockets_max)
 {
```

Comment:
I am really not sure that passing an IP parameter at that point is useful. The selected source IP normally depends on the destination and address family. It would probably be much simpler to concentrate on the selection of the second source address during the "probe" call.